tests/integration/security/authorization_test.go

```diff
@@ -940,118 +1009,121 @@ func TestAuthorization_Conditions(t *testing.T) {
-					podBWithIPB := apps.B.Match(echo.InCluster(t.Clusters()[i])).Match(echo.Namespace(nsB.Name()))[0]
-
-					args := map[string]string{
-						"NamespaceA": nsA.Name(),
-						"NamespaceB": nsB.Name(),
-						"NamespaceC": nsC.Name(),
-						"IpA":        getWorkload(podAWithIPA, t).Address(),
-						"IpB":        getWorkload(podBWithIPB, t).Address(),
-						"IpC":        IPC,
-						"PortC":      fmt.Sprintf("%d", portC),
-					}
-
-					policies := tmpl.EvaluateAllOrFail(t, args, file.AsStringOrFail(t, "testdata/authz/v1beta1-conditions.yaml.tmpl"))
-					t.Config().ApplyYAMLOrFail(t, "", policies...)
+			c := apps.C.Match(echo.Namespace(nsC.Name()))
+			vm := apps.VM.Match(echo.Namespace(nsA.Name()))
+			for _, dst := range []echo.Instances{c, vm} {
+				for i := 0; i < len(t.Clusters()); i++ {
+					t.NewSubTest(fmt.Sprintf("from %s to %s in %s",
+						t.Clusters()[i].StableName(), dst[0].Config().Service, dst[0].Config().Cluster.Name())).
+						Run(func(t framework.TestContext) {
+							src1 := apps.A.Match(echo.InCluster(t.Clusters()[i])).Match(echo.Namespace(nsA.Name()))
+							src2 := apps.B.Match(echo.InCluster(t.Clusters()[i])).Match(echo.Namespace(nsB.Name()))
+							var ipList string
+							for i := 0; i < len(dst); i++ {
+								ipList += "\"" + getWorkload(dst[i], t).Address() + "\","
+							}
+							ipLen := len(ipList)
+							ipList = ipList[:ipLen-1]
+							args := map[string]string{
+								"dst":           dst[0].Config().Service,
+								"dstIP":         ipList,
+								"dstNamespace":  dst[0].Config().Namespace.Name(),
+								"dstPort":       "8090",
+								"src1":          util.ASvc,
+								"src2":          util.BSvc,
+								"src1IP":        getWorkload(src1[0], t).Address(),
+								"src2IP":        getWorkload(src2[0], t).Address(),
+								"src1Namespace": nsA.Name(),
+								"src2Namespace": nsB.Name(),
+							}
 
-					for _, srcCluster := range t.Clusters() {
-						t.NewSubTest(fmt.Sprintf("From %s", srcCluster.StableName())).Run(func(t framework.TestContext) {
-							a := apps.A.Match(echo.InCluster(srcCluster).And(echo.Namespace(nsA.Name())))
-							b := apps.B.Match(echo.InCluster(srcCluster).And(echo.Namespace(nsB.Name())))
+							policies := tmpl.EvaluateAllOrFail(t, args, file.AsStringOrFail(t, "testdata/authz/v1beta1-conditions.yaml.tmpl"))
+							t.Config().ApplyYAMLOrFail(t, "", policies...)
 							callCount := 1
 							if t.Clusters().IsMulticluster() {
 								// so we can validate all clusters are hit
 								callCount = util.CallsPerCluster * len(t.Clusters())
 							}
-							newTestCase := func(from echo.Instance, path string, headers map[string]string, expectAllowed bool) rbacUtil.TestCase {
+							newTestCase := func(from echo.Instances, path string, headers map[string]string, expectAllowed bool) rbacUtil.TestCase {
+								if len(from) == 0 && t.Clusters().IsMulticluster() {
```

Comment:
How can `from` be empty?

Reply:
It seems that in some clusters there are no vm instances, I see crashes with error like `from[0]` is out of range.